### PR TITLE
ET-3432 add vpc link

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## What 🕵️ 🔍

- adds network load balancer (nlb) [tutorial](https://aws.amazon.com/de/blogs/networking-and-content-delivery/application-load-balancer-type-target-group-for-network-load-balancer/)
- connects nlb with application load balancer
- adds vpc link in api gateway
- connects vpc link with nlb [tutorial](https://docs.aws.amazon.com/apigateway/latest/developerguide/getting-started-with-private-integration.html)
- adjusts GET integration method to use the vpc link
- removes the hello world lambda

## References 📝 🔗

[ET-3432](https://xainag.atlassian.net/browse/ET-3432)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/stage/nlb/main.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+data "terraform_remote_state" "alb" {
+  backend = "s3"
+  config = {
+    bucket = "xayn-infrastructure"
+    key    = "stage/alb/terraform.tfstate"
+    region = "eu-central-1"
+  }
+}
+
+resource "aws_lb_target_group" "alb" {
```
Comment (jorge-cr-13) [inline aws/stage/nlb/main.tf:19]: Is it not better to use the `aws_lb_target_group` we already have for the `alb`? This one [ecs_services](https://github.com/xaynetwork/xayn_infrastructure/blob/master/aws/stage/alb/main.tf#L35)

File: aws/stage/nlb/main.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+resource "aws_lb" "vpc_link_alb" {
+  name               = "alb-load-balancer"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = data.terraform_remote_state.vpc.outputs.subnet_private_ids.*
+}
+
+resource "aws_lb_listener" "vpc_link_alb" {
```
Comment (jorge-cr-13) [inline aws/stage/nlb/main.tf:39]: Same here is not better to reference this one? [ecs_services](https://github.com/xaynetwork/xayn_infrastructure/blob/master/aws/stage/alb/main.tf#L43)

File: aws/stage/nlb/main.tf (hunk 1 above)
Comment (Robert-Steiner) [reply aws/stage/nlb/main.tf:19]: They look similar, but unfortunately differ in `target_type`

File: aws/stage/nlb/main.tf (hunk 2 above)
Comment (Robert-Steiner) [reply aws/stage/nlb/main.tf:39]: Here the `load_balancer_arn` would be different

**Review by jorge-cr-13 — APPROVED** [review]

**Pull request merged**